"./x.py run miri --stage 1" no longer works

This used to work fine at some point, but now leads to some strange errors:
```
$ ./x.py run miri --stage 1 -- src/tools/miri/tests/pass/hello.rs
Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.02s
Building stage0 library artifacts (x86_64-unknown-linux-gnu)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.07s
Building compiler artifacts (stage0 -> stage1, x86_64-unknown-linux-gnu)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.11s
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
Building stage0 tool lld-wrapper (x86_64-unknown-linux-gnu)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.06s
Building tool miri (stage0 -> stage1, x86_64-unknown-linux-gnu)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.07s
Building tool cargo-miri (stage0 -> stage1, x86_64-unknown-linux-gnu)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.07s
Building stage1 miri sysroot (x86_64-unknown-linux-gnu)
A sysroot for Miri is already available in `/home/r/src/rust/rustc.3/build/x86_64-unknown-linux-gnu/miri-sysroot`.
    Finished `release` profile [optimized + debuginfo] target(s) in 0.07s
     Running `build/x86_64-unknown-linux-gnu/stage0-tools/x86_64-unknown-linux-gnu/release/miri --target x86_64-unknown-linux-gnu src/tools/miri/tests/pass/hello.rs`
error[E0786]: found invalid metadata files for crate `std`
  |
  = note: corrupt metadata encountered in /home/r/src/rust/rustc.3/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-e9d12b2c234bc463.rlib
  = note: corrupt metadata encountered in /home/r/src/rust/rustc.3/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-e9d12b2c234bc463.so
  = note: corrupt metadata encountered in /home/r/src/rust/rustc.3/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-ec2a734cf29b8414.rlib

error: cannot find macro `eprintln` in this scope
 --> src/tools/miri/tests/pass/hello.rs:3:5
  |
3 |     eprintln!("Hello, error!");
  |     ^^^^^^^^

error: cannot find macro `println` in this scope
 --> src/tools/miri/tests/pass/hello.rs:2:5
  |
2 |     println!("Hello, world!");
  |     ^^^^^^^

error: requires `sized` lang_item

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0786`.
Build completed unsuccessfully in 0:00:01
```

--- a/src/bootstrap/src/core/build_steps/run.rs
+++ b/src/bootstrap/src/core/build_steps/run.rs
@@ -149,12 +149,14 @@ impl Step for Miri {
             &[],
         );
         miri.add_rustc_lib_path(builder);
-        // Forward arguments.
         miri.arg("--").arg("--target").arg(target.rustc_target_arg());
-        miri.args(builder.config.args());
 
         // miri tests need to know about the stage sysroot
-        miri.env("MIRI_SYSROOT", &miri_sysroot);
+        miri.arg("--sysroot").arg(miri_sysroot);
+
+        // Forward arguments. This may contain further arguments to the program
+        // after another --, so this must be at the end.
+        miri.args(builder.config.args());
 
         let mut miri = Command::from(miri);
         builder.run(&mut miri);

